File: P101 Same Tree/Solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isSameTree(self, p, q):
        """
        :type p: TreeNode
        :type q: TreeNode
        :rtype: bool
        """
        if p==None and q==None:
            return True
        elif p != None and q != None:
            if p.val == q.val:
                return self.isSameTree(p.left, q.left) & self.isSameTree(q.right, p.right)
        else:
            return False
        
        return False

File: P101 Same Tree/Solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        if(p==NULL&&q==NULL) 
            return true;
        else if(p != NULL && q != NULL) {
            if(p->val == q->val)
                return isSameTree(p->left, q->left) && isSameTree(q->right, p->right);
        }
        else
            return false;
        
        return false;
    }
};